lib/ethereumex/http_client.ex

```diff
@@ -9,27 +9,32 @@ defmodule Ethereumex.HttpClient do
     options = [hackney: [pool: :default]] ++ Config.http_options()
     url = Keyword.get(opts, :url) || Config.rpc_url()
 
-    with {:ok, response} <- HTTPoison.post(url, payload, headers, options),
-         %HTTPoison.Response{body: body, status_code: code} = response do
-      decode_body(body, code)
-    else
-      {:error, %HTTPoison.Error{reason: reason}} -> {:error, reason}
-      e -> {:error, e}
+    case HTTPoison.post(url, payload, headers, options) do
```

Comment:
:+1:

Reply:
this was also refactored because of credo complaints